packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts

```diff
@@ -0,0 +1,135 @@
+import {type Draft as WritableDraft} from '@reduxjs/toolkit';
+import {createReducer, FacetValueRequest} from '../../../../ssr.index';
+import {
+  toggleExcludeFacetValue,
+  toggleSelectFacetValue,
+  updateFacetIsFieldExpanded,
+  updateFacetNumberOfValues,
+} from '../../../facets/facet-set/facet-set-actions';
+import {convertFacetValueToRequest} from '../../../facets/facet-set/facet-set-slice';
+import {fetchProductListing} from '../../product-listing/product-listing-actions';
+import {
+  CommerceFacetSetState,
+  getCommerceFacetSetInitialState,
+} from './facet-set-state';
+import {CommerceFacetRequest} from './interfaces/request';
+import {AnyFacetResponse, FacetResponse} from './interfaces/response';
+
+export const commerceFacetSetReducer = createReducer(
+  getCommerceFacetSetInitialState(),
+
+  (builder) => {
+    builder
+      .addCase(fetchProductListing.fulfilled, (state, action) => {
+        const facets = action.payload.response.facets;
+        facets.forEach((facetResponse) =>
```

Comment:
I'd tend to use a for..of instead of .forEach, but that's mostly a personal preference